sim/battle.ts

```diff
@@ -1712,18 +1714,16 @@ export class Battle extends Dex.ModdedDex {
 	start() {
 		if (this.active) return;
 
-		if (!this.p1 || !this.p2) {
-			// need two players to start
-			return;
-		}
+		// need all players to start
+		if (!this.sides.every(side => !!side)) return;
```

Comment:
Lots of `!`s! Maybe `this.sides.some(side => !side)`?

Reply:
Why not `!this.sides.every(side => side)`

Reply:
`Array.every` makes a fuss about `side` not being a boolean.